fairlearn/reductions/moments/conditional_opportunity.py

```diff
@@ -48,6 +48,18 @@ def signed_weights(self, lambda_vec):
         )
         return signed_weights
 
+    def extract_A(self, X, aux_data):
+        if self.from_X:
+            if self.column_name is not None:
+                return X[self.column_name]
+            else:
+                raise NotImplementedError("This one is hard given range in input types")
+        else:
+            if self.column_name is not None:
+                return aux_data[self.column_name]
```

Comment:
aux_data may not be a dataframe, though

Reply:
Hence [Incomplete]. I was just trying to get a happy path working so we had something concrete to discuss